jetty-websocket/websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/WebSocketServerComponents.java

```diff
@@ -92,6 +97,8 @@ public static WebSocketComponents ensureWebSocketComponents(Server server, Servl
             serverComponents.unmanage(deflaterPool);
         if (server.contains(bufferPool))
             serverComponents.unmanage(bufferPool);
+        if (executor != null)
+            serverComponents.unmanage(executor);
```

Comment:
It's a little funny to create the WSSC with executor as auto and then externally force to unmanaged.   Kind of feels like the WSSC should be making these decisions for itself internally.    But if this is the only place the WSSCs are instantiated, then I guess it is OK?

Reply:
I'm following the same pattern as t he other components here.

Although I think it may not be necessary as I think the threadPool should always be started by the time we reach here, so it will not be managed by the WebSocketComponents. But I don't think there is any harm to have this line just to be sure.